presto-kafka07/src/main/java/com/facebook/presto/kafka/KafkaRecordSet.java

```diff
@@ -239,6 +246,43 @@ private boolean nextRow(MessageAndOffset messageAndOffset)
             return true; // Advanced successfully.
         }
 
+        private void openFetchRequest()
+        {
+            if (messageAndOffsetIterator == null) {
+                log.info("Fetching %d bytes from partition %d @offset %d (%d - %d) -- %d messages read so far",
+                        fetchSize, split.getPartitionId(), cursorOffset, split.getStart(), split.getEnd(), totalMessages);
+                cursorOffset += fetchedSize;
+                FetchRequest req = new FetchRequest(split.getTopicName(), split.getPartitionId(), cursorOffset, fetchSize);
```

Comment:
Can we check if the next cursorOffset is ahead of the split end, so you might save to download the last segment?